build.rs

```diff
@@ -0,0 +1,56 @@
+use rerun_except::rerun_except;
+
+use std::env;
+use std::path::{Path, PathBuf};
+use std::process::Command;
+
+static BOEHM_REPO: &str = "git://github.com/ivmai/bdwgc.git";
+static BOEHM_ATOMICS_REPO: &str = "git://github.com/ivmai/libatomic_ops.git";
+static BOEHM_DIR: &str = "bdwgc";
+static POINTER_MASK: &str = "-DPOINTER_MASK=0xFFFFFFFFFFFFFFF8";
+
+fn run<F>(name: &str, mut configure: F)
+where
+    F: FnMut(&mut Command) -> &mut Command,
+{
+    let mut command = Command::new(name);
+    let configured = configure(&mut command);
+    println!("Executing {:?}", configured);
+    if !configured.status().is_ok() {
+        panic!("failed to execute {:?}", configured);
+    }
+    println!("Command {:?} finished successfully", configured);
+}
+
+fn main() {
+    rerun_except(&["gc_tests/tests/*.rs"]).unwrap();
+
+    let mut bdwgc = PathBuf::from(env::var("CARGO_MANIFEST_DIR").unwrap());
+    bdwgc.push("target");
+    #[cfg(debug_assertions)]
+    bdwgc.push("debug");
+    #[cfg(not(debug_assertions))]
+    bdwgc.push("release");
+
+    let mut bdwgc_libs = PathBuf::from(&bdwgc);
+    bdwgc_libs.push("lib");
+
+
+    if bdwgc_libs.exists() {
+        return
+    }
+
+    if !Path::new(BOEHM_DIR).exists() {
+        run("git", |cmd| cmd.arg("clone").arg(BOEHM_REPO));
+
+        run("git", |cmd| {
+            cmd.arg("clone").arg(BOEHM_ATOMICS_REPO).current_dir(BOEHM_DIR)
+        });
+    }
+
+    env::set_current_dir(BOEHM_DIR).unwrap();
+
+    run("./autogen.sh", |cmd| cmd);
+    run("./configure", |cmd| cmd.arg(format!("--prefix={}", bdwgc.to_str().unwrap())).env("CFLAGS", POINTER_MASK));
```

Comment:
Since we're not running `make install`, I imagine that the `prefix` is irrelevant?

Reply:
We are using `make install` to put the shared objects in the `target/` dir.

Reply:
I'm an idiot. Sorry!

Reply:
I've actually changed this now. We use the `OUT_DIR` (which I wasn't aware of) to let cargo decide where to put the shared objects.